### DISCUSSION TITLE
使用AE，转成BMP导出一直显示失败，转成视频导出也显示黑屏

### DISCUSSION THREAD

**Discussion by karen00007** [Tencent/libpag#2156] [Q&A] [answered]
使用AE，转成BMP导出一直显示失败，转成视频导出也显示黑屏
![1708683543228](https://github.com/Tencent/libpag/assets/161021921/c6335f6f-f7ad-4ece-b823-c309f7c71fbb)

**Comment by createmomo** [comment]
你好，我也遇到了这个问题，请问后来解决了吗？

**Reply by avchenfu** [reply]
我们也遇到了这个问题

**Comment by avchenfu** [comment]
我们周边十几个人都遇到了这个问题，怎么官方都没人重视的，都没人来解决？

**Reply by avchenfu** [reply]
同样的AE软件版本，在少部分电脑可以正常导出PAG，但大部分电脑，无论怎么配置，都是一次不成功。感觉是电脑系统不兼容。

**Comment by 1111titi** [comment]
+1勾选BMP就导出失败

**Comment by createmomo** [comment]
我后来不得已找到了一个临时解决的办法，也不知道有没有什么损失。在pag config中，把BMP生成视频序列改成图片序列能导出了。 @1111titi @avchenfu

**Reply by avchenfu** [reply]
图片序列类似GIF图一样，很不流畅，文件也大

**Reply by createmomo** [reply]
我采纳了官方的另一个建议（采用历史版本），确实成功了。（从最新的4.3.6换到了4.0.5；windows）。

**Comment by whyisthishard** [comment]
我也遇到了，这么常见的问题怎么就是不优化嘞：(

**Comment by newstarchen** [comment]
你好，请提供一下版本号和失败的工程文件。

**Reply by newstarchen** [reply]
在此之前可以尝试使用历史版本看是否能成功。https://pag.art/docs/pag-download.html

**Reply by createmomo** [reply]
谢谢，因为项目原因不便分享工程文件。但是采纳了你的另一个建议（采用历史版本），确实成功了。（从最新的4.3.6换到了4.0.5；windows）。谢谢啦。

**Comment by jinwuwu001** [comment]
请问该问题出现的环境是否还在，若还在请帮忙确认如下问题
如下图所示，输入%APPDATA%\H264EncoderTools
<img width="421" alt="1" src="https://github.com/user-attachments/assets/0ec264eb-25e1-43b9-8222-aeb7189f4628" />

然后回车进入对应文件夹，检查下该文件夹下是否存在名为H264EncoderTools.exe的可执行程序
<img width="710" alt="2" src="https://github.com/user-attachments/assets/570b4bdf-7765-4c1a-9e91-e603a839f9cc" />

若不存在且此时AE导出bmp文件失败，请在此评论回复下便于确认问题原因

**Comment by jinwuwu001** [comment, marked as answer]
经确认该问题的原因是PAGViewer在进行ae插件安装时没有正确安装264编码工具，该问题是偶现的，目前正在处理中，后续版本会修复，可通过如下两种方式暂时规避该问题
1、可以在PAGViewer中多次安装ae插件，确保%APPDATA%\H264EncoderTools下存在编码工具后再进行pag导出
2、在PAGViewer安装目录下找到可执行程序PAGViewer/H264EncoderTools.exe，将其复制至%APPDATA%\H264EncoderTools，然后再进行pag文件导出
